src/coreclr/jit/lsrabuild.cpp

```diff
@@ -892,6 +892,10 @@ regMaskTP LinearScan::getKillSetForCall(GenTreeCall* call)
     if (!compiler->compFloatingPointUsed)
     {
         killMask &= ~RBM_FLT_CALLEE_TRASH;
+
+#if defined(TARGET_XARCH)
+        killMask &= ~RBM_MSK_CALLEE_TRASH;
+#endif // TARGET_XARCH
```

Comment:
Perhaps interestingly, having this vs not having it makes a difference to the integer registers allocated.

That, in general, seems odd and one would expect that the kill mask including or excluding floating/mask registers shouldn't impact how integer registers are allocated (particularly when there are no floating-point or mask registers actually in use).